src/timepicker-select/timepicker-select.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, Input, ChangeDetectorRef } from "@angular/core";
+import { Select } from "../select/select.component";
+
+@Component({
+	selector: "ibm-timepicker-select",
+	template: `
+		<div
+			[ngClass]="{
+				'bx--select--inline': display === 'inline',
+				'bx--select--light': theme === 'light',
+				'bx--skeleton': skeleton
+			}"
+			class="bx--select bx--time-picker__select">
+			<label *ngIf="!skeleton" [attr.for]="id" class="bx--label bx--visually-hidden">{{label}}</label>
+			<select
+				#select
+				[attr.id]="id"
+				[disabled]="disabled"
+				(change)="onChange($event)"
+				class="bx--select-input">
+				<ng-content></ng-content>
+			</select>
+			<svg *ngIf="!skeleton" class="bx--select__arrow" width="10" height="5" viewBox="0 0 10 5">
+			<path d="M0 0l5 4.998L10 0z" fill-rule="evenodd" />
+			</svg>
+		</div>
+	`,
+	styles: ["select { -webkit-appearance : none ; }"]
+})
+export class TimePickerSelect extends Select {
+
+	@Input() display: "inline" | "default" = "default";
+
+	@Input() id = `timepicker-select-${TimePickerSelect.selectCount++}`;
+	/**
+	 * Set to true to disable component.
+	 */
+	@Input() disabled = false;
+	/**
+	 * Set to true for a loading select.
+	 */
+	@Input() skeleton = false;
+
+	constructor(protected changeDetectorRef: ChangeDetectorRef) {
+		super();
+		TimePickerSelect.selectCount++;
```

Comment:
The constructor can be removed. As is, this would end up incrementing `TimePickerSelect.selectCount` twice